File: ly_thuyet/buoi6(contest)/to_hop.cpp
#include<bits/stdc++.h>

using namespace std;

#define mod 10000007
long long f[1200][1200];

void check() {
    int n, k;
    cin >> n >> k;
    if (k == 0 || k == 1) {
        cout << 1 << endl;
        return;
    }
    for (int i = 0; i <= k; i++) {
        for (int j = i; j <= n; j++) {
            if (i == 0 || i == j) 
                f[i][j] = 1;
            else 
                f[i][j] = (f[i - 1][j - 1] + f[i][j - 1]) % mod;
        }
    }
    cout << f[k][n] % mod << endl;
}
int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);
    int t = 1;
    cin >> t;
    while (t--) {
        check();
    }
}

File: ly_thuyet/buoi6(contest)/to_hop.py
mod = 10000007

store = [[0]*1100]*1100

def check():
    n, k = input().split()
    n, k = int(n), int(k)
    if k==0 or k==1:
        print(1)
        return
    for i in range(k+1):
        for j in range(i,n+1,1):
            if i==0 or i==j:
                store[i][j] = 1
            else:
                store[i][j] = (store[i - 1][j - 1] + store[i][j - 1]) % mod
    print(store[k][n]%mod)

t = int(input())

for u in range(t):
    check()